src/components/Comment/CommentText.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+interface Props {
+  text: string;
+  onClickTag: (type: string, user: string) => void;
+}
+
+export const CommentText = ({ text, onClickTag }: Props) => {
+  const tags = text.match(/@\[.*?\]\(user:.*?\)/gi) || [];
+  const otherText = text.split(/@\[.*?\]\(user:.*?\)/gi);
+
+  return (
+    <>
+      {otherText[0] || ""}
+      {tags.map((tag, idx) => {
+        // @ts-ignore:next-line
+        const tagDisplay = tag.match(/\[.*?\]/gi)[0]?.slice(1, -1);
+        // @ts-ignore:next-line
+        const tagId = tag.match(/\(user:.*?\)/gi)[0]?.slice(6, -1);
+
+        return (
+          <React.Fragment key={idx}>
+            {otherText[idx + 1] || ""}
+            <button
+              type="button"
+              key={tagId}
+              onClick={() => onClickTag("user", tagId)}
+              className="bg-blue-200 rounded"
+            >
+              {tagDisplay || ""}
```

Comment:
can we move this logic to the top where we are calculating tagDisplay value

Reply:
Curious, why is this is a concern? Moving it to top makes line longer and makes it part of ignore lint.

Reply:
As we are already do the value computation at the top, why do we need to handle value absence case in the jsx expression we can do that during the computation itself.

Reply:
A readability stylistic preference? It doesn't add confusion, and at least to make, makes it easier to read. Otherwise I have to parse a long line to understand the behavior.

Reply:
I generally prefer doing value calculation at once, UTY.